src/client/game.cpp

```diff
@@ -2060,8 +2060,9 @@ void Game::openInventory()
 	InventoryLocation inventoryloc;
 	inventoryloc.setCurrentPlayer();
 
-	if (!client->modsLoaded()
-			|| !client->getScript()->on_inventory_open(fs_src->m_client->getInventory(inventoryloc))) {
+	if ((!client->modsLoaded()
+			|| !client->getScript()->on_inventory_open(fs_src->m_client->getInventory(inventoryloc)))
+			&& fs_src->getForm() != "") {
```

Comment:
This can use some refactoring:
<pre>
if (!client->modsLoaded() || !client->getScript()->on_inventory_open(fs_src->m_client->getInventory(inventoryloc)))
    return;
if (fs_src->getForm().empty())
    return;
// the rest
</pre>
It also looks like `fs_src` is leaked if the inventory is not shown (it should be deleted).

Reply:
Changed. By `fs_src` being leaked do you mean it leaks memory? Should I delete the variable so the memory gets deallocated?

Reply:
Yes. In the normal case I assume `GUIFormSpecMenu::create` somehow takes care of it, but for every return statement there should be a `delete fs_src;` first.